### PR TITLE
new parent-contact screen

### PR DESCRIPTION
**Pull request by guiburato**
#### 🔗 Pivotal Tracker ticket
[187108109](https://www.pivotaltracker.com/story/show/187108109)

#### ✍️ Description
new parent-contact screen

#### 📷 Design reference
[Figma](https://www.figma.com/file/1oRFY7p6tdGogBJk7Ugle8/IL-CCAP-Full-Flow-(WIP)?type=design&node-id=1190-33950&mode=design&t=0UtL6jpz70cFDpRi-0)

#### ✅ Completion tasks

- [ x ] Added relevant tests
- [ x ] Meets acceptance criteria

### CODE REVIEW

File: src/main/java/org/ilgcc/app/inputs/Gcc.java [hunk 1]
```diff
@@ -34,6 +37,11 @@ public class Gcc extends FlowInputs {
     private String parentIsServing;
     private String parentInMilitaryReserveOrNationalGuard;
 
+    private String parentContactPhoneNumber;
+    private String parentContactEmail;
```
[the remark this answers is not in the document]
Comment (guiburato) [reply src/main/java/org/ilgcc/app/inputs/Gcc.java:41]: Done. :)

**Review by sree-cfa — CHANGES_REQUESTED** [review]
Very close - couple things to fix
- "How do you prefer to be communicated with?*" is required - there should be an error message if client tries to continue without it
- The validation is switched here. There should be an error if they select "paperless" and didnt provide and email. 
Screenshot 2024-03-04 at 5 17 33 PM

File: src/main/java/org/ilgcc/app/submission/actions/ValidateParentContactProvideEmail.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Locale locale = LocaleContextHolder.getLocale();
+    Map<String, List<String>> errorMessages = new HashMap<>();
+    Map<String, Object> inputData = formSubmission.getFormData();
+    if (!inputData.containsKey(INPUT_NAME_PREFERENCE)) {
+      return errorMessages;
+    }
+
+    String paperless = inputData.get(INPUT_NAME_PREFERENCE).toString();
+    if (paperless.isBlank()){
+      errorMessages.put(INPUT_NAME_PREFERENCE, List.of(messageSource.getMessage("general.indicates-required", null, locale)));
```
Comment (sree-cfa) [inline src/main/java/org/ilgcc/app/submission/actions/ValidateParentContactProvideEmail.java:42]: Our pattern is "Please answer..." for missing required fields. I think we can use this generic one

```suggestion
      errorMessages.put(INPUT_NAME_PREFERENCE, List.of(messageSource.getMessage("errors.required", null, locale)));
```

**Review by sree-cfa — APPROVED** [review]
made one suggestion - otherwise lgtm!

**Pull request merged**